usb/device/targets/TARGET_Freescale/USBPhy_Kinetis.cpp

```diff
@@ -317,6 +318,12 @@ void USBPhyHw::ep0_read(uint8_t *data, uint32_t size)
     } else {
         endpoint_read(EP0OUT, data, size);
     }
+
+    // Clear suspend after the setup stage
+    if (setup_suspend) {
+        USB0->CTL &= ~USB_CTL_TXSUSPENDTOKENBUSY_MASK;
```

Comment:
Is there a reason this register couldn't simply be read out instead of needing setup_suspend?

Reply:
Nvm, final blurb makes it all clear.